src/libstd/sys/common/thread_info.rs

```diff
@@ -57,6 +57,10 @@ pub fn set(stack_guard: Option<usize>, thread: Thread) {
 }
 
 // a hack to get around privacy restrictions; implemented by `std::thread`
+#[doc(hidden)]
+#[unstable(feature = "libstd_sys_internals",
+           reason = "this trait is considered exported for some reason",
```

Comment:
The "for some reason" here is that the set of exported items is pretty lossy when it comes to traits, although I can't seem to find the issue right now... Regardless it's ok to omit `doc(hidden)` here as they shouldn't be documented anyway.

It may be worth considering the intersection of public and exported items for traits perhaps?